go/api/request_routing_config.go

```diff
@@ -0,0 +1,191 @@
+	case SlotTypeReplica:
+		return protobuf.SlotTypes_Replica, nil
+	default:
+		return protobuf.SlotTypes_Primary, &RequestError{"Invalid slot type"}
+	}
+}
+
+// Request routing configuration overrides the [api.ReadFrom] connection configuration.
+// If SlotTypeReplica is used, the request will be routed to a replica, even if the strategy is ReadFrom [api.PreferReplica].
+type SlotIdRoute struct {
+	slotType SlotType
+	slotID   int32
+}
+
+// - slotType: Defines type of the node being addressed.
+// - slotId: Slot number. There are 16384 slots in a Valkey cluster, and each shard manages a slot range. Unless the slot is
+// known, it's better to route using [api.SlotTypePrimary].
+func NewSlotIdRoute(slotType SlotType, slotId int32) *SlotIdRoute {
+	return &SlotIdRoute{slotType: slotType, slotID: slotId}
+}
+
+func (slotIdRoute *SlotIdRoute) toRoutesProtobuf() (*protobuf.Routes, error) {
+	slotType, err := mapSlotType(slotIdRoute.slotType)
+	if err != nil {
+		return nil, err
+	}
+
+	request := &protobuf.Routes{
+		Value: &protobuf.Routes_SlotIdRoute{
+			SlotIdRoute: &protobuf.SlotIdRoute{
+				SlotType: slotType,
+				SlotId:   slotIdRoute.slotID,
+			},
+		},
+	}
+	return request, nil
+}
+
+// Request routing configuration overrides the [api.ReadFrom] connection configuration.
+// If SlotTypeReplica is used, the request will be routed to a replica, even if the strategy is ReadFrom [api.PreferReplica].
+type SlotKeyRoute struct {
+	slotType SlotType
+	slotKey  string
+}
+
+// - slotType: Defines type of the node being addressed.
+// - slotKey: The request will be sent to nodes managing this key.
+func NewSlotKeyRoute(slotType SlotType, slotKey string) *SlotKeyRoute {
+	return &SlotKeyRoute{slotType: slotType, slotKey: slotKey}
+}
+
+func (slotKeyRoute *SlotKeyRoute) toRoutesProtobuf() (*protobuf.Routes, error) {
+	slotType, err := mapSlotType(slotKeyRoute.slotType)
+	if err != nil {
+		return nil, err
+	}
+
+	request := &protobuf.Routes{
+		Value: &protobuf.Routes_SlotKeyRoute{
+			SlotKeyRoute: &protobuf.SlotKeyRoute{
+				SlotType: slotType,
+				SlotKey:  slotKeyRoute.slotKey,
+			},
+		},
+	}
+	return request, nil
+}
+
+// Routes a request to a node by its address.
+type ByAddressRoute struct {
+	host string
+	port int32
+}
+
+// Create a route using hostname/address and port.
+// - host: The endpoint of the node. If port is not provided, host should be in the "address:port" format, where address is the
+// preferred endpoint as shown in the output of the CLUSTER SLOTS command.
+// - port: The port to access the node. If port is not provided, host is assumed to be in the format "address:port".
+func NewByAddressRoute(host string, port int32) *ByAddressRoute {
+	return &ByAddressRoute{host: host, port: port}
+}
+
+// Create a route using address string formatted as "address:port".
+func NewByAddressRouteWithHost(host string) (*ByAddressRoute, error) {
+	split := strings.Split(host, ":")
+	if len(split) < 2 || len(split) > 2 {
```

Comment:
Is this condition the same as the following?  
```suggestion
	if len(split) != 2{
```